router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go

```diff
@@ -0,0 +1,606 @@
+package bingads
+
+import (
+	"archive/zip"
+	"bufio"
+	"encoding/csv"
+	"encoding/json"
+	stdjson "encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	bingads "github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/bing-ads/bingads_sdk"
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/common"
+	oauth "github.com/rudderlabs/rudder-server/services/oauth"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	"golang.org/x/oauth2"
+)
+
+type BingAdsBulkUploader struct {
+	destName             string
+	accessToken          string
+	developerToken       string
+	refreshToken         string
+	oauthClient          *oauth.OAuthErrResHandler
+	service              *bingads.BulkService
+	destinationIDFileMap map[string]string
+	timeout              time.Duration
+}
+
+type User struct {
+	Email       string `json:"email"`
+	HashedEmail string `json:"hashedEmail"`
+}
+type Message struct {
+	List []User `json:"List"`
+}
+type Metadata struct {
+	JobID int64 `json:"job_id"`
+}
+
+// This struct represent each line of the text file created by the batchrouter
+type Data struct {
+	Message  Message  `json:"message"`
+	Metadata Metadata `json:"metadata"`
+}
+
+type DestinationConfig struct {
+	AudienceId               string   `json:"audienceId"`
+	CustomerAccountId        string   `json:"customerAccountId"`
+	CustomerId               string   `json:"customerId"`
+	OneTrustCookieCategories []string `json:"oneTrustCookieCategories"`
+	RudderAccountId          string   `json:"rudderAccountId"`
+}
+
+type secretStruct struct {
+	AccessToken     string
+	RefreshToken    string
+	Developer_token string
+	ExpirationDate  string
+}
+
+type Response struct {
+	Status   string      `json:"status"`
+	Metadata MetadataNew `json:"metadata"`
+}
+
+type MetadataNew struct {
+	FailedKeys    []int64           `json:"failedKeys"`
+	FailedReasons map[string]string `json:"failedReasons"`
+	WarningKeys   []string          `json:"warningKeys"`
+	SucceededKeys []int64           `json:"succeededKeys"`
+}
+
+/*
+This function create zip file from the text file created by the batchrouter
+It takes the text file path as input and returns the zip file path
+The maximum size of the zip file is 100MB, if the size of the zip file exceeds 100MB then the job is marked as failed
+*/
+func createZipFile(filePath string, failedJobIds *[]int64, successJobIds *[]int64, audienceId string) string {
```

Comment:
idiomatic go `Id` -> `ID`, `Json` -> `JSON`

> Words in names that are initialisms or acronyms (e.g. “URL” or “NATO”) have a consistent case. For example, “URL” should appear as “URL” or “url” (as in “urlPony”, or “URLPony”), never as “Url”. As an example: ServeHTTP not ServeHttp. For identifiers with multiple initialized “words”, use for example “xmlHTTPRequest” or “XMLHTTPRequest”.



```suggestion
func createZipFile(filePath string, failedJobIDs *[]int64, successJobIds *[]int64, audienceID string) string {
```